src/Database/LSMTree.hs

```diff
@@ -1656,10 +1722,28 @@ _withInternalTables (Table (table :: Internal.Table m h) :| tables) action =
     | otherwise = throwIO $ ErrTableUnionHandleTypeMismatch 0 (typeRep $ Proxy @h) i (typeRep $ Proxy @h')
 
 {- |
-Get the amount of remaining union debt.
+Get an /upper bound/ for the amount of remaining union debt.
```

Comment:
Union debt is already defined to be an upper bound: 

> The amount of union debt is an upper bound on the amount of computation that
needs to be performed before an incremental union is completed.

So you could now read this as "an upper bound on an upper bound"

Same is true in a few other places in this module

Reply:
Fair, so we should redefine union debt, because debt as a concept is a number not an upper bound.

Reply:
Fixed in 4c07e5d16d0a7072e292c2ed11bed6487838eed5